libsolidity/codegen/YulUtilFunctions.h

```diff
@@ -250,10 +251,16 @@ class YulUtilFunctions
 	/// Return value: pointer
 	std::string allocationFunction();
 
-	/// @returns the name of a function that allocates a memory array.
+	/// @returns the name of a function that allocates a memory array
+	/// and zeroes it if _shouldZero is true.
 	/// For dynamic arrays it adds space for length and stores it.
 	/// signature: (length) -> memPtr
-	std::string allocateMemoryArrayFunction(ArrayType const& _type);
+	std::string allocateMemoryArrayFunction(ArrayType const& _type, bool _shouldZero = true);
```

Comment:
Is this ever used with `_shouldZero == false`? Shouldn't we always zero for security reasons?

Reply:
Right now the default is `true`, so it's never used with `false`. Agree.